### ISSUE TITLE
Minify not works when page is cached and minify cache is expired (*.old)

### ISSUE THREAD

**Issue by nigrosimone** [szepeviktor/fix-w3tc#287]
**TL;DR**

Minify cache not works with `+Multiviews`

**In details (long story)**

Using disk cache for both page and minify, when a page is cached and a minify cache is expired:
- on 0.9.5.x, the cached page is displayed unstyled and unscripted;
- on 0.9.4.x, all works as expected.

**Step to reproduce**

1) set page cache and minify cache on "disk" and logout.
2) generate the cache with viewing a "test" page.
3) go into the directory `\wp-content\cache\minify\0\ ` and **rename all file** by add a suffix `.old`
4) view the "test" page, now unstyled.

[![fx1.jpg](https://s24.postimg.org/y09un8k2t/fx1.jpg)](https://postimg.org/image/xnigh21sx/)
[![fx2.jpg](https://s28.postimg.org/m6d8fkqnh/fx2.jpg)](https://postimg.org/image/mvw0rxr6x/)

**Strange facts**

1) if you go into the directory `\wp-content\cache\minify\0\ ` and **delete all file** w3tc regenerates minify cache and all works, it's ok, but garbage collector for minify cache, rename expired file with suffix `.old`.

**My environments**

- Apache 2.4
- PHP 5.6.24 / php5_module
- SO Windows Vista 32bit (dev. machine) 
- Wordpress 4.7.1
- Theme Avada 5.6
- tested both W3TC-FIX 0.9.5.x master and W3TC 0.9.5.1 (wp.org)
- W3TC settings in [master.json](https://gist.github.com/nigrosimone/29d56788a0858469d60f11fa872b99dc)

**Updates**

1) i found another issue #288, but this fix not solve my problem.

2) in Apache error log i found 

> [Sun Jan 15 19:31:39.808370 2017] [negotiation:error] [pid 7248:tid 2264] [client 127.0.0.1:60075] AH00687: Negotiation: discovered file(s) matching request: C:/DevEnv/htdocs/www.wordpress.local/wp-content/cache/minify/1/985ff.js (None could be negotiated).

now... my vhosts is (see +Multiviews)

```
# www.wordpress.local #
<VirtualHost 127.0.0.2:*>
    DocumentRoot "C:/DevEnv/htdocs/www.wordpress.local"
    ServerName www.wordpress.local
	
    <Directory C:/DevEnv/htdocs/www.wordpress.local>
        DirectoryIndex index.php
        Options -Indexes +FollowSymLinks +MultiViews
        AllowOverride All
        Require all granted
    </Directory>
</VirtualHost>
```
from [docs](https://httpd.apache.org/docs/2.4/mod/mod_negotiation.html):

>  A Multiviews search (enabled by the Multiviews Options), where the server does an implicit filename pattern match, and choose from amongst the results.

i have this setting because there are the same setting of my hosting (very famous italian hosting), i have changed `+MultiViews` to `-MultiViews` and it works!

w3tc v0.9.5.x add `-MultiViews` only if compatibility mode is checked [Minify_Environment.php#L542](https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.5.x/Minify_Environment.php#L542):

```php
$rules .= W3TC_MARKER_BEGIN_MINIFY_CACHE . "\n";
if ( $compatibility ) {
    $rules .= "Options -MultiViews\n";
}
```
but v0.9.4.x add it always [MinifyAdminEnvironment.php#L539](https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.4.x/lib/W3/MinifyAdminEnvironment.php#L539):
```php
$rules .= W3TC_MARKER_BEGIN_MINIFY_CACHE . "\n";
$rules .= "Options -MultiViews\n";
```
but why this change? `mod_negotiation` is a [default module of Apache](https://en.wikipedia.org/wiki/List_of_Apache_modules)....

My proposal is revert to v0.9.4.x's approach with a tiny control:

```php
$rules .= W3TC_MARKER_BEGIN_MINIFY_CACHE . "\n";
$rules .= "<IfModule mod_negotiation.c>\n";
$rules .= "    Options -MultiViews\n";
$rules .= "</IfModule>\n";
```
My final PR https://github.com/szepeviktor/w3-total-cache-fixed/pull/289

I think i'm right, but i'm not a web-server expert and i want a community review from @amiga-500, @charlesLF, @szepeviktor.

_side note: i have deleted all comments because not helpful for understand the issue or off topic._

**Comment by vijayaraghavanramanan** [comment]
Does the folder have a .htaccess file? I think it's the one generating new minified files if they don't exist.

**Comment by nigrosimone** [comment]
> Does the folder have a .htaccess file? I think it's the one generating new minified files if they don't exist.

yes, \wp-content\cache\minify\.htaccess

```
# BEGIN W3TC Minify cache
FileETag MTime Size
<IfModule mod_mime.c>
    AddEncoding gzip .gzip
    <Files *.css.gzip>
        ForceType text/css
    </Files>
    <Files *.js.gzip>
        ForceType application/x-javascript
    </Files>
</IfModule>
<IfModule mod_deflate.c>
    <IfModule mod_setenvif.c>
        SetEnvIfNoCase Request_URI \.gzip$ no-gzip
    </IfModule>
</IfModule>
<IfModule mod_expires.c>
    ExpiresActive On
    ExpiresByType text/css M31536000
    ExpiresByType application/x-javascript M31536000
</IfModule>
<IfModule mod_headers.c>
    Header set X-Powered-By "W3 Total Cache/0.9.5.1"
    Header set Vary "Accept-Encoding"
    Header set Pragma "public"
    Header append Cache-Control "public"
</IfModule>
# END W3TC Minify cache
# BEGIN W3TC Minify core
<IfModule mod_rewrite.c>
    RewriteEngine On
    RewriteBase /wp-content/cache/minify/
    RewriteCond %{HTTP:Accept-Encoding} gzip
    RewriteRule .* - [E=APPEND_EXT:.gzip]
    RewriteCond %{REQUEST_FILENAME}%{ENV:APPEND_EXT} -f
    RewriteRule (.*) $1%{ENV:APPEND_EXT} [L]
    RewriteRule ^(.+\.(css|js))$ /index.php [L]
</IfModule>
# END W3TC Minify core
```

**Comment by vijayaraghavanramanan** [comment]
I think the older version had this: 

```
# BEGIN W3TC Minify core
<IfModule mod_rewrite.c>
    RewriteEngine On
    RewriteBase /wp-content/cache/minify/
    RewriteRule /w3tc_rewrite_test$ ../../plugins/w3-total-cache/pub/minify.php?w3tc_rewrite_test=1 [L]
    RewriteCond %{HTTP:Accept-Encoding} gzip
    RewriteRule .* - [E=APPEND_EXT:.gzip]
    RewriteCond %{REQUEST_FILENAME}%{ENV:APPEND_EXT} -f
    RewriteRule (.*) $1%{ENV:APPEND_EXT} [L]
    RewriteRule ^(.+/[X]+\.css)$ ../../plugins/w3-total-cache/pub/minify.php?test_file=$1 [L]
    RewriteRule ^(.+\.(css|js))$ ../../plugins/w3-total-cache/pub/minify.php?file=$1 [L]
</IfModule>
# END W3TC Minify core
```

If you observe it refers to the file minify.php in the plugins folder. This did the minify on the fly. Seems to be missing from the latest version.

**Comment by nigrosimone** [comment]
@vijayaraghavanramanan 0.9.5.x refers to the file index.php (wordpress):

    RewriteRule ^(.+\.(css|js))$ /index.php [L]

but not seems to works.

**Comment by vijayaraghavanramanan** [comment]
Yeah why should it refer to that file which doesn't have anything to do with W3 Total Cache? It's a Wordpress file, so can't minify. Needs to be changed, right?

**Comment by nigrosimone** [comment]
> Yeah why should it refer to that file which doesn't have anything to do with W3 Total Cache? It's a Wordpress file, so can't minify. Needs to be changed, right?

good question! but i don't have the response

**Comment by szepeviktor** [comment]
Please use GitHub labels.

**Comment by nigrosimone** [comment]
> Please use GitHub labels.

@szepeviktor you can create a label for 0.9.4.x?

**Comment by szepeviktor** [comment]
Oops.
